geode-cq/src/distributedTest/java/org/apache/geode/security/MultiUserAPIDUnitTest.java

```diff
@@ -252,4 +283,50 @@ public void jsonFormatterOnTheClientWithMultiUser() throws Exception {
     PdxInstance value = regionService.getJsonFormatter().toPdxInstance(json);
     regionView.put("key", value);
   }
+
+  @Test
+  public void transactionTest() throws Exception {
+    ClientCache cache = client.withServerConnection(server.getPort())
+        .withCredential("data", "data")
+        .createCache();
+    Region<Object, Object> region = client.createProxyRegion("region");
+    CacheTransactionManager txManager =
+        cache.getCacheTransactionManager();
+
+    try {
+      txManager.begin();
+      region.put("key1", "value1");
+      region.put("key2", "value2");
+      txManager.commit();
+    } catch (CommitConflictException conflict) {
+      // ... do necessary work for a transaction that failed on commit
+    } finally {
+      if (txManager.exists()) {
+        txManager.rollback();
+      }
+    }
+  }
+
+  @Test
+  public void multiUserWithClientSubscription() throws Exception {
+    ClientCache cache = client.withServerConnection(server.getPort())
+        .withPoolSubscription(true)
+        .withProperty(SECURITY_CLIENT_AUTH_INIT, CountableUserPasswordAuthInit.class.getName())
+        .withMultiUser(true)
+        .createCache();
+    Properties properties = new Properties();
+    properties.setProperty(UserPasswordAuthInit.USER_NAME, "data");
+    properties.setProperty(UserPasswordAuthInit.PASSWORD, "wrongPassword");
+    RegionService regionService = cache.createAuthenticatedView(properties);
+
+    cache.createClientRegionFactory(ClientRegionShortcut.PROXY).create("region");
+    Region region = regionService.getRegion(SEPARATOR + "region");
+    assertThatThrownBy(() -> region.registerInterestForAllKeys())
+        .isInstanceOf(UnsupportedOperationException.class);
```

Comment:
This assertion seems unrelated to the code being tested in this class, since `registerInterest()` will always throw an `UnsupportedOperationException` when called on a `ProxyRegion`, regardless of the authentication being used. Can it just be removed?

Reply:
the test was added for code understanding. Not needed.